src/DevAdventCalendarCompetition/DevAdventCalendarCompetition.Repository/TestRepository.cs

```diff
@@ -25,33 +27,49 @@ public Test GetById(int testId)
             return this._dbContext.Set<Test>().FirstOrDefault(el => el.Id == testId);
         }
 
-        public void UpdateDates(Test test)
+        public Test GetByNumber(int testNumber)
```

Comment:
if you are using `AddTest` and `GetTestsWithUserAnswers` then you should rename this to `GetTestByNumber`